### ISSUE TITLE
Save as dialog should gray out 'Delete' and maybe 'Share'

### ISSUE THREAD

**Issue by ToonTalk** [jmoenig/Snap--Build-Your-Own-Blocks#2133]
When selecting save as the dialog has all the usual save options:

![image](https://user-images.githubusercontent.com/2138696/41960177-5e3abe64-79ef-11e8-941b-4a6bcb1882da.png)

It makes no sense to Delete and currently Share does nothing so it either should be gray or should share and save (see #63).

**Comment by cycomachead** [comment]
Requires #649 

FWIW, I think we should just hide them. Why even have people try to read a low contrast button if it can't do anything.

**Comment by ToonTalk** [comment]
I found this debate about hiding versus disabling buttons: https://www.quora.com/When-is-it-preferable-to-hide-a-disabled-button-in-an-interface-instead-of-showing-it-in-a-disabled-state#

I didn't see what I remember learning which is that people who use an interface frequently often subconsciously use the layout and ignore labels. So disappearing buttons changes the layout and someone may click the wrong button because it is where they are used to finding a different button.

**Comment by jmoenig** [comment]
FWIW (TIL this, LOL) the "delete" and "share" buttons are there because they do work. Before you're saving a project, while you're at it, you can totally share one or more other ones, and also delete projects you no longer want. This does not affect the currently opened project, which you can still save.

**Comment by cycomachead** [comment]
Lol 😀

But that’s a really dangerous thing for a save dialog to support!
Clicking on a project updates the file name,  does it not? That means sharing another project makes it easy to overwrite it if you aren’t careful. 

-- 
Michael Ball
From my iPhone
michaelball.co

**Comment by jmoenig** [comment]
That's right, Michael. But before actually overwriting an existing project the project dialog is warning you about it and lets you decide whether you really want to.
I don't mind hiding those buttons, though. On the subject of "ghosted" vs. hidden I'm leaning towards hiding things that have no functionality instead of showing something that isn't available.

**Comment by ToonTalk** [comment]
I think this topic applies as well to the editor when tracing is on when some buttons are missing. #1853 

I just found this [Button UX Design: Best Practices, Types and States](https://uxplanet.org/button-ux-design-best-practices-types-and-states-647cf4ae0fc6)

> Arguments for hiding the button:
> 
> Clarity. Only showing what is needed for the task at hand.
> 
> Save estate. It allows you to change the controls, using the same space for different means. This is handy when there is a lot going on. 
> 
> Arguments for using a disabled state:
> 
> Show the action possibility. Even if button isn’t in use, the user has a chance to learn that the action is possible. You may even have a tooltip explaining the criteria for use.
> 
> Control location. The user can learn where controls and buttons live within the interface.

My personal preference is disabled buttons with tool tip explanations.

**Comment by brianharvey** [comment]
Graying out would be the right choice if a button is sometimes useful and sometimes not useful in the same dialog.  That's not the case here.  The DELETE button was never a good idea here, and the SHARE button should always have been SAVE AND SHARE.

I don't think DELETE or SHARE belong in the Open dialog, either.  That was a historical kludge to get around our lack of a web site integrated with the cloud.  Now we have that, and so deleting projects should be done from the "my stuff" page, not from the editor at all.  Sharing a project should mostly be done there, too, but I support SAVE AND SHARE in the Save As dialog as a convenient shortcut.

P.S.  Just a reminder that we've all agreed to remove the Browser button on the left, and instead have a button Local Disk that closes our Open/Save dialog and opens the OS one.  (Keeping Browser in the Open dialog for a while, or maybe forever if we detect localstore projects.)

**Comment by bromagosa** [comment]
> Now we have that,

Woah, hold your horses! Now we're _working_ on that. In Catalan we say "_don't say "wheat" until it's all been bagged and tied up_" ;)

I agree these should be handled in the site, but we should keep them there for now until we finish the site and go live.

**Comment by brianharvey** [comment]
Yes, yes, of course.  But we shouldn't go down the road of redesigning those dialogs without remembering the end goal.  :-)
